### PR TITLE
[CIS-801] Websocket events audit

### PR DESCRIPTION
**Pull request by DominikBucher12**
# What this PR do:
- Audits Websocket Events and ensures that they are parsed correctly against Backend Payloads 
# How to test it
- Run the tests and check if the events are received and parsed correctly

### CODE REVIEW

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift [hunk 1]
```diff
@@ -72,7 +72,7 @@ struct ChannelPayload<ExtraData: ExtraDataTypes>: Decodable {
     }
 }
 
-struct ChannelDetailPayload<ExtraData: ExtraDataTypes>: Decodable {
+public struct ChannelDetailPayload<ExtraData: ExtraDataTypes>: Decodable {
```
Comment (VojtaStavik) [inline Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift:75]: This shouldn't be public

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift [hunk 2]
```diff
@@ -257,26 +257,25 @@ public struct ChannelConfig: Codable {
...
+        searchEnabled = try container.decode(Bool.self, forKey: .searchEnabled)
+        mutesEnabled = try container.decode(Bool.self, forKey: .mutesEnabled)
+        urlEnrichmentEnabled = try container.decode(Bool.self, forKey: .urlEnrichmentEnabled)
+        messageRetention = try container.decode(String.self, forKey: .messageRetention)
+        maxMessageLength = try container.decode(Int.self, forKey: .maxMessageLength)
+        commands = try container.decodeIfPresent([Command].self, forKey: .commands) ?? []
+        
+        createdAt = try container.decode(Date.self, forKey: .createdAt)
+        updatedAt = try container.decode(Date.self, forKey: .updatedAt)
+    }
```
Comment (VojtaStavik) [inline Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift:278]: Why this change?

File: Sources/StreamChat/APIClient/Endpoints/Payloads/CurrentUserPayloads_Tests.swift [hunk 3]
```diff
@@ -35,7 +35,7 @@ class CurrentUserPayload_Tests: XCTestCase {
         XCTAssertEqual(payload.mutedChannels[0].user.id, "broken-waterfall-5")
         XCTAssertEqual(payload.mutedChannels[0].mutedChannel.cid.rawValue, "messaging:B1DFF9C5-E6A6-4BFA-9375-DC5E8C6852FF")
         XCTAssertEqual(payload.mutedChannels[0].createdAt, "2021-03-22T10:23:52.516225Z".toDate())
-        XCTAssertEqual(payload.mutedChannels[0].updatedAt, "2021-04-22T10:23:52.516225Z".toDate())
+        XCTAssertEqual(payload.mutedChannels[0].updatedAt, "2021-03-22T10:23:52.516225Z".toDate())
```
Comment (VojtaStavik) [inline Sources/StreamChat/APIClient/Endpoints/Payloads/CurrentUserPayloads_Tests.swift:38]: This looks suspicious

File: Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift [hunk 4]
```diff
@@ -27,7 +27,7 @@ enum UserPayloadsCodingKeys: String, CodingKey {
 // MARK: - GET users
 
 /// An object describing the incoming user JSON payload.
-class UserPayload<ExtraData: UserExtraData>: Decodable {
+public class UserPayload<ExtraData: UserExtraData>: Decodable {
```
Comment (VojtaStavik) [inline Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift:30]: this should stay internal

File: Sources/StreamChat/WebSocketClient/Events/EventPayload.swift [hunk 5]
```diff
@@ -7,7 +7,7 @@ import Foundation
 // MARK: - Temporary
 
 /// The DTO object mirroring the JSON representation of an event.
-struct EventPayload<ExtraData: ExtraDataTypes>: Decodable {
+public struct EventPayload<ExtraData: ExtraDataTypes>: Decodable {
```
Comment (VojtaStavik) [inline Sources/StreamChat/WebSocketClient/Events/EventPayload.swift:10]: should be internal

**Review by VojtaStavik — CHANGES_REQUESTED** [review]

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift (hunk 1 above)
Comment (DominikBucher12) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift]: Fixed.

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift (hunk 2 above)
Comment (DominikBucher12) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift:278]: For debugging purposes it was better to have this to figure out what is going on. Anyway I deleted it and we can always put it back.

File: Sources/StreamChat/APIClient/Endpoints/Payloads/CurrentUserPayloads_Tests.swift (hunk 3 above)
Comment (DominikBucher12) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/CurrentUserPayloads_Tests.swift:38]: But it is not. It was copy paste error in the `Channel` object in `muted_channel` vs the `muted_channel` object which contains both `user` and `channel`. This is fix, this should already fail in https://github.com/GetStream/stream-chat-swift/pull/1011   


```
"muted_channel" : {
.... 
"channel":{ "createdAt" : RightDate }
"createdAt" : WrongDateUsedAtTests
}```

File: Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift (hunk 4 above)
Comment (DominikBucher12) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift]: Fixed.

File: Sources/StreamChat/WebSocketClient/Events/EventPayload.swift (hunk 5 above)
Comment (DominikBucher12) [reply Sources/StreamChat/WebSocketClient/Events/EventPayload.swift]: Fixed.

File: Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift (hunk 2 above)
Comment (VojtaStavik) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift:252]: @DominikBucher12 if you think we need it, let's keep it. Your call :)

Comment (DominikBucher12) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/ChannelListPayload.swift:278]: okay 180 turn, this is actually very needed 😅  Otherwise the swift-generated `init(decoder:)` fails to decode some of the fields and throws errors.

**Review by VojtaStavik — APPROVED** [review]
